.github/workflows/sync_to_gitlab.yml

```
@@ -0,0 +1,39 @@
+# SPDX-FileCopyrightText: 2025 DESY and the Constellation authors
+# SPDX-License-Identifier: CC0-1.0
+
+name: Sync PRs to DESY GitLab
+
+on:
+  pull_request_target:
+    types: [opened, synchronize]
+  schedule:
+    - cron: '0 * * * *'  # Runs every hour
+
+jobs:
+  sync:
+    runs-on: ubuntu-latest
+    env:
+      GITLAB_REPOSITORY: "gitlab.desy.de/constellation/constellation"
+      GITLAB_PROJECT_ID: ${{ secrets.GITLAB_PROJECT_ID }}
+      GITLAB_PROJECT_TOKEN: ${{ secrets.GITLAB_PROJECT_TOKEN }}
+      BRANCH_PREFIX: "github-"
+
+    steps:
+      - name: Checkout PR branch
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event.pull_request.head.ref }}
+
+      - name: Push branch to GitLab
+        env:
+          PR_BRANCH: ${{ github.event.pull_request.head.ref }}
+
+        run: |
+          git remote add gitlab "https://$GITLAB_REPOSITORY.git"
+          git push --force "https://oauth2:$GITLAB_TOKEN@$GITLAB_REPOSITORY.git" "$PR_BRANCH:$BRANCH_PREFIX$PR_BRANCH"
```

```suggestion
          git push gitlab --force "https://oauth2:$GITLAB_TOKEN@$GITLAB_REPOSITORY.git" "$GITHUB_HEAD_REF:$BRANCH_PREFIX-$GITHUB_ACTOR-$GITHUB_HEAD_REF"
```
Add the github user in case two users use the same branch name